python/cudf/cudf/core/dataframe.py

```diff
@@ -2217,14 +2125,16 @@ def transpose(self):
         """
         # Never transpose a MultiIndex - remove the existing columns and
         # replace with a RangeIndex. Afterward, reassign.
-        temp_columns = self.columns.copy(deep=False)
-        self.columns = pd.Index(range(len(self.columns)))
-        result = libcudf.transpose.transpose(self)
-        self.columns = temp_columns
-        result = result.rename(dict(zip(result.columns, self.index)))
-        result = result.set_index(temp_columns)
-        if isinstance(self.index, cudf.core.multiindex.MultiIndex):
-            result.columns = self.index
+        inp = self.copy(deep=False)
+        temp_columns = inp.columns.copy(deep=False)
+        temp_index = inp.index.copy(deep=False)
+        if len(self._data) == 0:
+            return DataFrame(index=temp_columns, columns=temp_index)
+        inp.columns = pd.RangeIndex(start=0, stop=len(self.columns))
+        inp.index = RangeIndex(start=0, stop=len(self))
+        result = libcudf.transpose.transpose(inp)
+        result._index = as_index(temp_columns)
+        result.columns = temp_index
```

Comment:
Should we explicitly copy `temp_index` to Pandas here? Otherwise it does it implicitly via iteration which is bad.

Reply:
Fixed